src/amo/components/SearchFilters/index.js

```diff
@@ -231,7 +236,7 @@ export class SearchFiltersBase extends React.Component {
             id="SearchFilters-OperatingSystem"
             name="operatingSystem"
             onChange={this.onSelectElementChange}
-            value={filters.operatingSystem || NO_FILTER}
+            value={convertOSToFilterValue(userAgentInfo.os.name)}
```

Comment:
I am surprised that this does not break any test cases. I guess we should still use `filters.operatingSystem` if available, though. (and maybe fallback to `NO_FILTER` if `convertOSToFilterValue(userAgentInfo.os.name)` is undefined)

Reply:
`filters.operatingSystem` isn't working for any Operating System, should I give it one more try?
Although, @tofumatt suggested to use `convertOSToFilterValue(userAgentInfo.os.name)` here: https://github.com/mozilla/addons-frontend/issues/3169#issuecomment-330222104 , if I understood it correctly.

Reply:
well, `filters.operatingSystem` should work when a user uses the `select`, no?

Reply:
Ah, I see. Let me fix it.